src/environments.js

```diff
@@ -85,8 +88,10 @@ function defineEnvironment(names, props, handler) {
 // is part of the source2e.pdf file of LaTeX2e source documentation.
 defineEnvironment("array", {
     numArgs: 1
-}, function(pos, mode, envName, colalign, positions) {
-    var parser = this;
+}, function(context, args) {
+    var colalign = args[0];
+    var lexer = context.lexer;
+    var positions = context.positions;
```

Comment:
Since `lexer` and `positions` are only used once we may as well use `context.lexer` and `context.positions` directly.